tests/test_datasets.py

```diff
@@ -1,13 +1,73 @@
 """This module contains the tests for the DatasetsClient."""
 import os
-from axiom import Client
+import unittest
+from logging import getLogger
+from .helpers import get_random_name
+from axiom import Client, DatasetCreateRequest, DatasetUpdateRequest
 
 
-def test_ingest():
-    """Tests the ingest endpoint"""
-    client = Client(os.getenv("AXIOM_DEPLOYMENT_URL"), os.getenv("AXIOM_TOKEN"))
-    print(
-        client.datasets.ingest(
-            os.getenv("AXIOM_DATASET"), [{"foo": "bar"}, {"bar": "baz"}]
+class TestDatasets(unittest.TestCase):
+
+    dataset_name: str
+    client: Client
+
+    @classmethod
+    def setUpClass(cls):
```

Comment:
A teardown that checks if the dataset still exists and deletes it would be great, otherwise we might run into zombie datasets on failures.

Reply:
makes sense, on it.

Reply:
updated